src/DefaultEditor.js

```diff
@@ -64,6 +66,18 @@ class DefaultEditor extends Component {
     return this.context.fullData.some(d => traceHasColorbar({}, d));
   }
 
+  hasLegend() {
+    return this.context.fullData.some(t => t.showlegend !== undefined); // eslint-disable-line no-undefined
```

Comment:
are we sure that this still enables us to press the 'showlegend' button, I think that in some cases its hidden by default no?

Reply:
This hides the panel only if there are absolutely no traces that support `showlegend`. So all should be fine.